lib/helpers.rb

```diff
@@ -153,6 +156,17 @@ def handle_threads_query(
       comment_threads = context_threads
     end
 
+    if author_id
+      comment_threads = comment_threads.where(:author_id => author_id)
+      if author_id != user_id
+        comment_threads = comment_threads.where(:anonymous => false, :anonymous_to_peers => false)
+      end
+    end
+
+    if thread_type
+      comment_threads = comment_threads.where({:thread_type => thread_type})
+    end
```

Comment:
If someone is requesting their own threads, we can return their anonymous posts. However don't return anonymous posts when filtering by another author, since that will essentially leak the author information.